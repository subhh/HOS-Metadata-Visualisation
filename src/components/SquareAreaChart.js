import React, { Fragment, useMemo } from 'react'
import { sum, values, entries, orderBy, sumBy } from 'lodash'
import { withTheme } from 'styled-components'
import styled from 'styled-components'
import { Box } from 'rebass'
import { ChartContainer, Title, Grid, Cell } from './ChartAtoms'
import ChartLegend from './ChartLegend'

const Wrapper = styled(Box)`
  border-radius: 4px;
  position: relative;
  overflow: hidden;

  @supports (mix-blend-mode: soft-light) {
    &:before {
      content: 'Â ';
      position: absolute;
      opacity: 0.3;
      background-image: linear-gradient(183deg, #ffffff 0%, #000000 100%);
      mix-blend-mode: soft-light;
      width: 100%;
      height: 100%;
    }
  }
`

/**
 * Converts values to part-to-whole data, visualizes data with square area chart
 */
const SquareAreaChart = ({
  size = 10,
  data,
  title,
  theme,
  flipColors,
  reverse,
}) => {
  const colors = useMemo(
    () =>
      flipColors ? theme.chartColors.slice().reverse() : theme.chartColors,
    [flipColors, theme.chartColors]
  )

  const quantities = useMemo(() => {
    const total = sum(values(data))
    const pairs = entries(data).map(([key, value]) => {
      const percentage = (value / total) * (size * size) || 0
      const rounded = Math.round(percentage)
      return [key, { percentage, rounded }]
    })

    const sorted = orderBy(pairs, '[1].percentage', 'desc')
    const highestVal = sorted[0]
    // if shares do not sum up to 100 we correct the highest value to keep the error minimal
    const diff = 100 - sumBy(sorted, '[1].rounded')
    if (diff && highestVal) {
      highestVal[1].rounded += diff
    }
    // order values by key
    return orderBy(pairs, 0, [reverse ? 'desc' : 'asc'])
  }, [data, reverse, size])

  return (
    <ChartContainer>
      {title && <Title>{title}</Title>}
      <Wrapper mb={4}>
        <Grid size={size} data-testid="grid">
          {quantities.map(([key, { rounded }], catIdx) => (
            <Fragment key={`group-${catIdx}`}>
              {Array.apply(null, Array(rounded)).map((v, j) => (
                <Cell
                  data-testid="cell"
                  key={`cell-${catIdx}-${j}`}
                  bg={colors[catIdx]}
                />
              ))}
            </Fragment>
          ))}
        </Grid>
      </Wrapper>
      <ChartLegend
        data={quantities.map(([key, value]) => key)}
        reverse={flipColors}
      />
    </ChartContainer>
  )
}

export default withTheme(SquareAreaChart)
