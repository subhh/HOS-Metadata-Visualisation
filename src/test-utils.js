import React, { useReducer } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { MemoryRouter } from 'react-router-dom'
import { StateContext, DispatchContext, reducer, initialState } from './state'
import theme from './theme'

const AllTheProviders = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState)

  return (
    <ThemeProvider theme={theme}>
      <StateContext.Provider value={state}>
        <DispatchContext.Provider value={dispatch}>
          {children}
        </DispatchContext.Provider>
      </StateContext.Provider>
    </ThemeProvider>
  )
}

const WithRouter = ({ children }) => (
  <MemoryRouter>
    <AllTheProviders>{children}</AllTheProviders>
  </MemoryRouter>
)

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

export const customRenderWithRouter = (ui, options) =>
  render(ui, {
    wrapper: WithRouter,
    ...options,
  })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }

export const mockAsync = data => {
  jest.spyOn(global, 'fetch').mockImplementation(url => {
    return Promise.resolve({
      json: () => Promise.resolve(data),
    })
  })
}

export const restoreAsyncMock = () => {
  global.fetch.mockRestore()
}
