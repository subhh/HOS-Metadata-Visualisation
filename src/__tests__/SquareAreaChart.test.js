import React from 'react'
import { render } from 'test-utils'
import SquareAreaChart from 'components/SquareAreaChart'

const props = {
  data: { var1: 10, var2: 30 },
  title: 'Title',
}

it('should match snapshot without data', () => {
  const { container } = render(<SquareAreaChart />)
  expect(container).toMatchSnapshot()
})

it('should match snapshot with data provided', () => {
  const { container } = render(<SquareAreaChart {...props} />)
  expect(container).toMatchSnapshot()
})

it('should match snapshot with flipped colors', () => {
  const { container } = render(<SquareAreaChart flipColors {...props} />)
  expect(container).toMatchSnapshot()
})

it('should match snapshot with reversed order', () => {
  const { container } = render(<SquareAreaChart reverse {...props} />)
  expect(container).toMatchSnapshot()
})

it('number of children should always sum up to square of size', () => {
  const { getAllByTestId } = render(
    <SquareAreaChart data={{ var1: 1, var2: 199 }} />
  )
  expect(getAllByTestId('cell').length).toBe(100)
})

it('number of children should always sum up to square of size', () => {
  const { getAllByTestId } = render(
    <SquareAreaChart data={{ var1: 1, var2: 1, var3: 1 }} />
  )
  expect(getAllByTestId('cell').length).toBe(100)
})
