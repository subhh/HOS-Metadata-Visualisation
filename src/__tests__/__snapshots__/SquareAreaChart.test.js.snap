// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot with data provided 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-htoDjs bZvMhP"
  >
    <h2
      class="sc-bdVaJa sc-htpNat sc-bxivhb sc-dnqmqq fCKnkB"
      color="blueDark"
      font-size="2"
      font-weight="500"
    >
      Title
    </h2>
    <div
      class="sc-bdVaJa sc-eNQAEJ dChLpc"
    >
      <div
        class="sc-bdVaJa sc-jzJRlG kHObkz"
        data-testid="grid"
        size="10"
      >
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
      </div>
    </div>
    <ul
      class="sc-ckVGcZ eHdLWN"
    >
      <li
        class="sc-jKJlTe ItYRm"
        font-size="0"
      >
        var1
      </li>
      <li
        class="sc-jKJlTe hqaXqp"
        font-size="0"
      >
        var2
      </li>
    </ul>
  </div>
</div>
`;

exports[`should match snapshot with flipped colors 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-htoDjs bZvMhP"
  >
    <h2
      class="sc-bdVaJa sc-htpNat sc-bxivhb sc-dnqmqq fCKnkB"
      color="blueDark"
      font-size="2"
      font-weight="500"
    >
      Title
    </h2>
    <div
      class="sc-bdVaJa sc-eNQAEJ dChLpc"
    >
      <div
        class="sc-bdVaJa sc-jzJRlG kHObkz"
        data-testid="grid"
        size="10"
      >
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
      </div>
    </div>
    <ul
      class="sc-ckVGcZ eHdLWN"
    >
      <li
        class="sc-jKJlTe hqaXqp"
        font-size="0"
      >
        var1
      </li>
      <li
        class="sc-jKJlTe ItYRm"
        font-size="0"
      >
        var2
      </li>
    </ul>
  </div>
</div>
`;

exports[`should match snapshot with reversed order 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-htoDjs bZvMhP"
  >
    <h2
      class="sc-bdVaJa sc-htpNat sc-bxivhb sc-dnqmqq fCKnkB"
      color="blueDark"
      font-size="2"
      font-weight="500"
    >
      Title
    </h2>
    <div
      class="sc-bdVaJa sc-eNQAEJ dChLpc"
    >
      <div
        class="sc-bdVaJa sc-jzJRlG kHObkz"
        data-testid="grid"
        size="10"
      >
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG iEplYp"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
        <div
          class="sc-cSHVUG joKvVR"
          data-testid="cell"
        />
      </div>
    </div>
    <ul
      class="sc-ckVGcZ eHdLWN"
    >
      <li
        class="sc-jKJlTe ItYRm"
        font-size="0"
      >
        var2
      </li>
      <li
        class="sc-jKJlTe hqaXqp"
        font-size="0"
      >
        var1
      </li>
    </ul>
  </div>
</div>
`;

exports[`should match snapshot without data 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-htoDjs bZvMhP"
  >
    <div
      class="sc-bdVaJa sc-eNQAEJ dChLpc"
    >
      <div
        class="sc-bdVaJa sc-jzJRlG kHObkz"
        data-testid="grid"
        size="10"
      />
    </div>
    <ul
      class="sc-ckVGcZ eHdLWN"
    />
  </div>
</div>
`;
